Can't build emacs --HEAD

`brew install emacs --HEAD` failing with the following error:

`
configure: error: You do not seem to have makeinfo >= 4.13, and your
source tree does not seem to have pre-built manuals in the 'info' directory.
`

I have brew-installed texinfo, even tried force link it (although it's keg-only formula), now 

`makeinfo --version` shows: `texi2any (GNU texinfo) 6.3`

I've tried removing Xcode and Xcode devtools and reinstalling them again - it didn't help.

However if I clone it manually from `git://git.savannah.gnu.org/emacs.git`, configure and build - it builds. 
So, something must have changed that broke the formula. Can someone take a look please?
ver. 25.1 of Emacs still builds (without --HEAD flag), but I'm afraid if they merge their changes then the stable would break as well

--- a/Formula/emacs.rb
+++ b/Formula/emacs.rb
@@ -22,6 +22,7 @@ class Emacs < Formula
 
     depends_on "autoconf" => :build
     depends_on "automake" => :build
+    depends_on "texinfo" => :build
   end
 
   option "with-cocoa", "Build a Cocoa version of emacs"

